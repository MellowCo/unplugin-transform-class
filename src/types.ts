import type { FilterPattern } from '@rollup/pluginutils'

export interface Options {
  /**
   * 自定义转换规则
   * @default
   * {
      '.': '-dl-',
      '/': '-sl-',
      ':': '-cl-',
      '%': '-pes-',
      '!': '-el-',
      '#': '-wn-',
      '(': '-lbl-',
      ')': '-lbr-',
      '[': '-lfl-',
      ']': '-lfr-',
      '$': '-do-',
      ',': '-lco-',
      '=': '-eqe-',
    }
   */
  rules?: Record<string, string>

  /**
   * 排除转换目标
   * @default [/[\\/]node_modules[\\/]/, /[\\/]\.git[\\/]/]
   */
  exclude?: FilterPattern

  /**
   * 需要转换的目标
   * @default [/\.[jt]sx?$/, /\.vue$/,  /\.vue\?vue/]
   */
  include?: FilterPattern
}
