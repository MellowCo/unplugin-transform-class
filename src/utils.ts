export const defaultRules: Record<string, string> = {
  '.': '_dl_',
  '/': '_sl_',
  ':': '_cl_',
  '%': '_pes_',
  '!': '_el_',
  '#': '_wn_',
  '(': '_lbl_',
  ')': '_lbr_',
  '[': '_lfl_',
  ']': '_lfr_',
  '$': '_do_',
  ',': '_lco_',
  '=': '_eqe_',
  '+': '_plus_',
  '*': '_star_',
}

export function escapeRegExp(str = '') {
  return str.replace(/[.*+?^${}()|[\]\\]/g, '\\$&')
}

function createTransformRegExp(rules: Record<string, string>, needEscape = false) {
  const escapePrefix = '\\'
  return new RegExp(Object.keys(rules).map(rule => escapeRegExp(`${needEscape ? escapePrefix : ''}${rule}`)).join('|'), 'g')
}

/**
 * 转换选择器字符
 * @param selector 选择器
 * @param rules 转换规则
 * @example bg-[#452233]:50 => bg--fl--w-452233-fr--c-40-c-50
 */
export function transformSelector(selector = '', rules = defaultRules) {
  const transformRegExp = createTransformRegExp(rules)

  return selector.replaceAll(transformRegExp, (m) => {
    return rules[m]
  })
}

export const cacheTransformSelector = (function () {
  let transformRegExp: RegExp

  return (selector = '', rules = defaultRules) => {
    if (!transformRegExp)
      transformRegExp = createTransformRegExp(rules)

    return selector.replaceAll(transformRegExp, (m) => {
      return rules[m]
    })
  }
}())

/**
 * 转换转义的选择器字符
 * @param selector 转义选择器
 * @param rules 转换规则
 * @example bg-\[\#452233\]\:50 => bg--fl--w-452233-fr--c-40-c-50
 */
export function transformEscapESelector(selector = '', rules = defaultRules) {
  const transformRegExp = createTransformRegExp(rules, true)

  return selector.replaceAll(transformRegExp, (m) => {
    return rules[m.replace('\\', '')]
  })
}

export const cacheTransformEscapESelector = (function () {
  let transformRegExp: RegExp

  return (selector = '', rules = defaultRules) => {
    if (!transformRegExp)
      transformRegExp = createTransformRegExp(rules, true)

    return selector.replaceAll(transformRegExp, (m) => {
      return rules[m.replace('\\', '')]
    })
  }
}())

/**
 * 还原转换后的选择器
 * @param selector 选择器
 * @param rules 转换规则
 * @example .bg--fl--w-452233-fr--c-40-c-50 => .bg-[#452233]:50
 */
export function restoreSelector(selector = '', rules = defaultRules) {
  const reverseRules = Object.fromEntries(Object.entries(rules).map(([key, value]) => [value, key]))
  const transformRegExp = createTransformRegExp(reverseRules)

  return selector.replaceAll(transformRegExp, (m) => {
    return reverseRules[m]
  })
}

export const cacheRestoreSelector = (function () {
  let transformRegExp: RegExp
  let reverseRules: Record<string, string>

  return (selector = '', rules = defaultRules) => {
    if (!transformRegExp) {
      reverseRules = Object.fromEntries(Object.entries(rules).map(([key, value]) => [value, key]))
      transformRegExp = createTransformRegExp(reverseRules)
    }

    return selector.replaceAll(transformRegExp, (m) => {
      return reverseRules[m]
    })
  }
}())

export { transformCode, getClass } from './core'
