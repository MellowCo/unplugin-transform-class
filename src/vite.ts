import unplugin from '.'

export { transformSelector } from './core'

export const transformWxClass = unplugin.vite
