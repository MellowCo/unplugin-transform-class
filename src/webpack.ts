import type { Options } from './types'
import unplugin from '.'

// TODO: some upstream lib failed generate invalid dts, remove the any in the future
export default unplugin.webpack as (options?: Options) => any
