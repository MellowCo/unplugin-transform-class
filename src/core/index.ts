import { trim } from '@meoc/utils'
import { cacheTransformSelector, defaultRules } from '../utils'

/**
 * 获取class
 */
export function getClass(code: string) {
  const matchs: string[][] = []
  // vue
  Array.from(code.matchAll(/\s:?[A-Za-z0-9]*[c|C]lass="([\s\S]*?)"/g)).forEach((m) => {
    const classStr = m[1]
    const sourceStr = trim(m[0])

    let classArr = [sourceStr]

    // 是否为动态class :class
    if (sourceStr.startsWith(':')) {
      if (classStr.startsWith('{'))
        classArr = classArr.concat(getObjClass(classStr))
      else if (classStr.startsWith('['))
        classArr = classArr.concat(getArrClass(classStr))
    }
    else {
      classArr.push(classStr)
    }

    matchs.push(classArr)
  })

  // react className="xxxx"
  Array.from(code.matchAll(/\s:?[A-Za-z0-9]*[c|C]lassName=["']([\s\S]*?)["']/g)).forEach((m) => {
    matchs.push([m[0], m[1]])
  })

  // className={xxxx}
  Array.from(code.matchAll(/\s:?[A-Za-z0-9]*[c|C]lassName=[{]([\s\S]*?)[}]/g)).forEach((m) => {
    matchs.push([m[0], ...Array.from(m[1].matchAll(/["']([\s\S]+?)["']/g)).map(v => v[1])])
  })

  return matchs
}

export function getObjClass(className: string) {
  // class="{ 'bg-teal-200:55': title === 'Hello', 'h-1.000%': title === 'Hello' }"
  // => ['bg-teal-200:55','h-1.000%']
  return Array.from(className.matchAll(/'([^']+?)'\s*:/g)).map(v => v[1])
}

export function getArrClass(className: string) {
  // [
  //   title === '2.3' ? 'font-$font-name bg-teal-200:55' :'tracking-[2/5]',
  //   isFont ? 'font-$font-name' : 'tracking-[2/5]'
  // ]
  // => ['font-$font-name bg-teal-200:55', 'tracking-[2/5]','font-$font-name', 'tracking-[2/5]']
  return Array.from(className.matchAll(/(?<=[\?\:&])\s*'(.*?)'/g)).map(v => v[1])
}

export function transformCode(code: string, rules = defaultRules) {
  const classNames = getClass(code)

  classNames.forEach((c) => {
    let currentClass = c[0]
    c.slice(1).forEach((selector) => {
      currentClass = currentClass.replace(selector, cacheTransformSelector(selector, rules))
    })
    code = code.replace(c[0], currentClass)
  })

  return code
}
