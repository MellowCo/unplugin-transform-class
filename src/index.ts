import { createUnplugin } from 'unplugin'
import type { Options } from './types'
import { transformCode } from './core'

export default createUnplugin<Options>(() => ({
  name: 'unplugin-transform-we-class',
  enforce: 'pre',
  transformInclude(id) {
    return /.(vue|jsx|tsx)$/.test(id)
  },
  transform(code) {
    return transformCode(code)
  },
}))
