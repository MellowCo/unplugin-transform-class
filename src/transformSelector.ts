import {transformSelector} from './core'

export default transformSelector
