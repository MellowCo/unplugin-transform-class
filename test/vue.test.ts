import { describe, expect, it } from 'vitest'
import { getArrClass, getClass, getObjClass, transformCode } from '../src/core'
import { restoreSelector } from '../src/utils'
import { equalCode, firstUI, getArrClassCode, getClassCode, getObjClassCode1, getObjClassCode2, replaceAllCode, tmText, vueCode, vueCode2, vueCode3, vueCode4, vueCode5, vueCode6 } from './assets/vue'

const rules = {
  '.': '-dr-',
  '/': '-sr-',
  ':': '-cr-',
  '%': '-pr-',
  '!': '-er-',
  '#': '-wr-',
  '(': '-blr-',
  ')': '-brr-',
  '[': '-flr-',
  ']': '-frr-',
  '$': '-rr-',
  ',': '-ccc',
  '=': '-eqq-',
}

describe('vue', () => {
  it('getArrClass', () => {
    expect(getArrClass(getArrClassCode)).toMatchSnapshot()
  })

  it('getObjClass', () => {
    expect(getObjClass(getObjClassCode1)).toMatchSnapshot()
    expect(getObjClass(getObjClassCode2)).toMatchSnapshot()
  })

  it('getClass', () => {
    expect(getClass(getClassCode)).toMatchSnapshot()
    expect(getClass(tmText)).toMatchSnapshot()
  })

  it('replaceAll', () => {
    expect(transformCode(replaceAllCode)).toMatchSnapshot()
  })

  it('transformCode', () => {
    expect(transformCode(vueCode)).toMatchSnapshot()
    expect(transformCode(vueCode2)).toMatchSnapshot()
    expect(transformCode(vueCode3)).toMatchSnapshot()
    expect(transformCode(vueCode4)).toMatchSnapshot()
    expect(transformCode(vueCode5)).toMatchSnapshot()
    expect(transformCode(vueCode6)).toMatchSnapshot()
    expect(transformCode(tmText)).toMatchSnapshot()
    expect(transformCode(firstUI)).toMatchSnapshot()
    expect(transformCode(equalCode)).toMatchSnapshot()
  })

  it('transform rules', () => {
    expect(transformCode(vueCode, rules)).toMatchSnapshot()
    expect(transformCode(vueCode2, rules)).toMatchSnapshot()
    expect(transformCode(vueCode3, rules)).toMatchSnapshot()
    expect(transformCode(vueCode4, rules)).toMatchSnapshot()
    expect(transformCode(vueCode5, rules)).toMatchSnapshot()
    expect(transformCode(tmText, rules)).toMatchSnapshot()
    expect(transformCode(firstUI, rules)).toMatchSnapshot()
    expect(transformCode(equalCode, rules)).toMatchSnapshot()
  })

  it('restoreSelector', () => {
    expect(restoreSelector(undefined, rules)).toMatchSnapshot()
    expect(restoreSelector('tracking--flr-2-sr-5-frr-', rules)).toMatchSnapshot()
    expect(restoreSelector('bg--flr-url-blr-https-cr--sr--sr-img-dr-cdn-dr-sugarat-dr-top-sr-mdImg-sr-MTY2ODA4OTc3MjcyMg-eqq--eqq-unocss-icon-gray-dr-svg-brr--frr-', rules)).toMatchSnapshot()
  })
})
