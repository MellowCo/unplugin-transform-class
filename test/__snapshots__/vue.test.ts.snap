// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`vue > getArrClass 1`] = `
[
  "font-$font-name bg-teal-200:55",
  "tracking-[2/5]",
  "font-$font-name",
  "tracking-[2/5]",
]
`;

exports[`vue > getClass 1`] = `
[
  [
    "class="text-red-500:20  bg-blue-200:20"",
    "text-red-500:20  bg-blue-200:20",
  ],
]
`;

exports[`vue > getClass 2`] = `
[
  [
    ":class="[fontSize ? '' : 'text-size-m', customClass]"",
    "",
    "text-size-m",
  ],
]
`;

exports[`vue > getObjClass 1`] = `
[
  "bg-teal-200:55",
  "h-1.000%",
  "font-$font-name",
]
`;

exports[`vue > getObjClass 2`] = `
[
  "bg-blue-600:80",
  "text-green-600/40",
]
`;

exports[`vue > replaceAll 1`] = `
"<view class="text-red-500_cl_20  bg-blue-200_cl_20 bg-_lfl__wn_333_lfr__sl_50 bg-_lfl__wn_999_lfr__sl_50">
text-red-500:20
</view>

<view xxclass="text-red-500_cl_20  bg-blue-200_cl_20 bg-_lfl__wn_333_lfr__sl_50 bg-_lfl__wn_999_lfr__sl_50">
text-red-500:20
</view>

<view xxClass="text-red-500_cl_20  bg-blue-200_cl_20 bg-_lfl__wn_333_lfr__sl_50 bg-_lfl__wn_999_lfr__sl_50">
text-red-500:20
</view>

"
`;

exports[`vue > restoreSelector 1`] = `""`;

exports[`vue > restoreSelector 2`] = `"tracking-[2/5]"`;

exports[`vue > restoreSelector 3`] = `"bg-[url(https://img.cdn.sugarat.top/mdImg/MTY2ODA4OTc3MjcyMg==unocss-icon-gray.svg)]"`;

exports[`vue > transform rules 1`] = `
"<template>
<view class="content-ccc1 text">
  <view class="indent-1-sr-2">
    indent-1/2
  </view>
  <view class="font--flr-system-ui-frr- bg-teal-200-cr-55">
    font-[system-ui]
  </view>
  <view class="font--rr-font-name">
    font-$font-name
  </view>

  <view class="font--rr-font-name" :class="[title === '2.3' ? 'font--rr-font-name bg-teal-200-cr-55' : 'tracking--flr-2-sr-5-frr-',isFont ? 'font--rr-font-name' : 'tracking--flr-2-sr-5-frr-']">
    [title === '2.3' ? 'font-$font-name bg-teal-200:55' : 'tracking-[2/5]',isFont ? 'font-$font-name' : 'tracking-[2/5]']
  </view>

  <view :class="{ 'bg-teal-200-cr-55': title === 'Hello', 'h-1-dr-000-pr-': title === 'Hello', 'font--rr-font-name': isFont }">
    { 'bg-teal-200:55': title === 'Hello', 'h-1.000%': title === 'Hello', 'font-$font-name': isFont }
  </view>

  <view class="tracking--flr-2-sr-5-frr- bg-teal-200-cr-55">
    tracking-[2/5]
  </view>

  <view class="h-1-dr-000-pr-">
    h-1.000%
  </view>

  <view class="h-1-dr-000-pr-">
  h-1.000%
</view>

  <view class="bg-teal-200-cr-55">
    bg-teal-200:55
  </view>
  <image class="logo" src="/static/logo.png" />
  <view class="text-area">
    <text class="title">
      {{ title }}
    </text>
  </view>
</view>
</template>"
`;

exports[`vue > transform rules 2`] = `
"
<template>
<view class="p-2">
  <view class="w-300 p-2 bg-green-200 text-center" @click="onChange">
    change flag
  </view>

  <view
    :class="{
      'bg-blue-600-cr-80': flag,
      'text-green-600-sr-40': !flag,
    }"

    :xxclass="{
      'bg-blue-600-cr-80': flag,
      'text-green-600-sr-40': !flag,
    }"

    :xxClass="{
      'bg-blue-600-cr-80': flag,
      'text-green-600-sr-40': !flag,
    }"
  >
    111111111
  </view>

  <view
    :class="{'bg-blue-600-cr-80': flag,'text-green-600-sr-40': !flag}"
  >
    111111111
  </view>


  <view
  :class="[
    flag ? 'bg-blue-600-sr-40' : 'bg-blue-600-cr-80',
    !flag ? 'text-yellow-800-sr-80' : 'text-yellow-800-sr-40',
  ]"
>
  111111111
</view>

<view
  :class="[
    flag ? 'bg-blue-600-sr-40' : 'bg-blue-600-cr-80',
    !flag ? 'text-yellow-800-sr-80' : 'text-yellow-800-sr-40',
  ]"
>
  111111111
</view>

<view class="content title foot">view</view>
</view>
</template>
"
`;

exports[`vue > transform rules 3`] = `
"
<template>
<view v-if="n.name=='img'" :class="'_img '+n.attrs.class" :style="n.attrs.style" :data-attrs="n.attrs" @tap="imgtap">
</view>
<view :class="n.attrs.class"></view>  
</template>
"
`;

exports[`vue > transform rules 4`] = `
"
<view class=""
>1. xxxxxxxx。
</view>

<view class="">2. 22222222。 </view>
"
`;

exports[`vue > transform rules 5`] = `
"
<uni-transition
v-if="maskShow"
:mode-class="['fade']"
:styles="maskClass"
:duration="duration"
:show="showTrans"
@click="onTap"
/>
<uni-transition
:mode-class="ani"
:styles="transClass"
:duration="duration"
:show="showTrans"
@click="onTap"
>
<view
  class="uni-popup__wrapper-box"
  :class="{isBottom: type == 'bottom','text-center':true}"
  @click.stop="clear"
>
  <slot />
</view>
</uni-transition>
"
`;

exports[`vue > transform rules 6`] = `
"
<text @click="emits('click', $event)" 
:selectable="selectable" 
:user-select="selectable" 
:class="[fontSize ? '' : 'text-size-m', customClass]" :style="[
  {
    lineHeight: (_fontSize ? _fontSize * 1.3 : 42) + props.unit, color: textColor
  },
  _fontSize ? { fontSize: _fontSize + props.unit } : '',
  customCSSStyle,
]">
<slot>{{ _label }}</slot>
</text>"
`;

exports[`vue > transform rules 7`] = `
"
<text :style="{ color: color, fontSize: getSize, fontWeight: fontWeight}" class="fui-icon" :class="[!color && !primary?'fui-icon__color':'',primary && !color?'fui-icon__active-color':'',disabled?'fui-icon__not-allowed':'',customPrefix,customPrefix?name:'']"
@click="handleClick">{{ icons[name] || '' }}</text>
"
`;

exports[`vue > transform rules 8`] = `
"<view class="bg--flr-url-blr-https-cr--sr--sr-img-dr-cdn-dr-sugarat-dr-top-sr-mdImg-sr-MTY2ODA4OTc3MjcyMg-eqq--eqq-unocss-icon-gray-dr-svg-brr--frr-">
bg ==
</view>"
`;

exports[`vue > transformCode 1`] = `
"<template>
<view class="content_lco_1 text">
  <view class="indent-1_sl_2">
    indent-1/2
  </view>
  <view class="font-_lfl_system-ui_lfr_ bg-teal-200_cl_55">
    font-[system-ui]
  </view>
  <view class="font-_do_font-name">
    font-$font-name
  </view>

  <view class="font-_do_font-name" :class="[title === '2.3' ? 'font-_do_font-name bg-teal-200_cl_55' : 'tracking-_lfl_2_sl_5_lfr_',isFont ? 'font-_do_font-name' : 'tracking-_lfl_2_sl_5_lfr_']">
    [title === '2.3' ? 'font-$font-name bg-teal-200:55' : 'tracking-[2/5]',isFont ? 'font-$font-name' : 'tracking-[2/5]']
  </view>

  <view :class="{ 'bg-teal-200_cl_55': title === 'Hello', 'h-1_dl_000_pes_': title === 'Hello', 'font-_do_font-name': isFont }">
    { 'bg-teal-200:55': title === 'Hello', 'h-1.000%': title === 'Hello', 'font-$font-name': isFont }
  </view>

  <view class="tracking-_lfl_2_sl_5_lfr_ bg-teal-200_cl_55">
    tracking-[2/5]
  </view>

  <view class="h-1_dl_000_pes_">
    h-1.000%
  </view>

  <view class="h-1_dl_000_pes_">
  h-1.000%
</view>

  <view class="bg-teal-200_cl_55">
    bg-teal-200:55
  </view>
  <image class="logo" src="/static/logo.png" />
  <view class="text-area">
    <text class="title">
      {{ title }}
    </text>
  </view>
</view>
</template>"
`;

exports[`vue > transformCode 2`] = `
"
<template>
<view class="p-2">
  <view class="w-300 p-2 bg-green-200 text-center" @click="onChange">
    change flag
  </view>

  <view
    :class="{
      'bg-blue-600_cl_80': flag,
      'text-green-600_sl_40': !flag,
    }"

    :xxclass="{
      'bg-blue-600_cl_80': flag,
      'text-green-600_sl_40': !flag,
    }"

    :xxClass="{
      'bg-blue-600_cl_80': flag,
      'text-green-600_sl_40': !flag,
    }"
  >
    111111111
  </view>

  <view
    :class="{'bg-blue-600_cl_80': flag,'text-green-600_sl_40': !flag}"
  >
    111111111
  </view>


  <view
  :class="[
    flag ? 'bg-blue-600_sl_40' : 'bg-blue-600_cl_80',
    !flag ? 'text-yellow-800_sl_80' : 'text-yellow-800_sl_40',
  ]"
>
  111111111
</view>

<view
  :class="[
    flag ? 'bg-blue-600_sl_40' : 'bg-blue-600_cl_80',
    !flag ? 'text-yellow-800_sl_80' : 'text-yellow-800_sl_40',
  ]"
>
  111111111
</view>

<view class="content title foot">view</view>
</view>
</template>
"
`;

exports[`vue > transformCode 3`] = `
"
<template>
<view v-if="n.name=='img'" :class="'_img '+n.attrs.class" :style="n.attrs.style" :data-attrs="n.attrs" @tap="imgtap">
</view>
<view :class="n.attrs.class"></view>  
</template>
"
`;

exports[`vue > transformCode 4`] = `
"
<view class=""
>1. xxxxxxxx。
</view>

<view class="">2. 22222222。 </view>
"
`;

exports[`vue > transformCode 5`] = `
"
<uni-transition
v-if="maskShow"
:mode-class="['fade']"
:styles="maskClass"
:duration="duration"
:show="showTrans"
@click="onTap"
/>
<uni-transition
:mode-class="ani"
:styles="transClass"
:duration="duration"
:show="showTrans"
@click="onTap"
>
<view
  class="uni-popup__wrapper-box"
  :class="{isBottom: type == 'bottom','text-center':true}"
  @click.stop="clear"
>
  <slot />
</view>
</uni-transition>
"
`;

exports[`vue > transformCode 6`] = `
"
<view class="bg-_lfl_--l-tab-bar-bg-color_lco_theme_lbl_backgroundColor_dl_container_lbr__lfr_" />
<view :class="{'bg-_lfl_--l-tab-bar-bg-color_lco_theme_lbl_backgroundColor_dl_container_lbr__lfr_': true}" />
<view :class="{ 'left-_lfl_calc_lbl_-1_star_var_lbl_--td-badge-basic-height_lbr__plus_1rpx_lbr__lfr_': 1 === 1 }">
test
</view>

<view :class="[1 === 1 ? 'left-_lfl_calc_lbl_-1_star_var_lbl_--td-badge-basic-height_lbr__plus_1rpx_lbr__lfr_ left' : ''] ">
  test
</view>

<view :class="[1 === 1 && 'left-_lfl_calc_lbl_-1_star_var_lbl_--td-badge-basic-height_lbr__plus_1rpx_lbr__lfr_ left'] ">
  test
</view>
"
`;

exports[`vue > transformCode 7`] = `
"
<text @click="emits('click', $event)" 
:selectable="selectable" 
:user-select="selectable" 
:class="[fontSize ? '' : 'text-size-m', customClass]" :style="[
  {
    lineHeight: (_fontSize ? _fontSize * 1.3 : 42) + props.unit, color: textColor
  },
  _fontSize ? { fontSize: _fontSize + props.unit } : '',
  customCSSStyle,
]">
<slot>{{ _label }}</slot>
</text>"
`;

exports[`vue > transformCode 8`] = `
"
<text :style="{ color: color, fontSize: getSize, fontWeight: fontWeight}" class="fui-icon" :class="[!color && !primary?'fui-icon__color':'',primary && !color?'fui-icon__active-color':'',disabled?'fui-icon__not-allowed':'',customPrefix,customPrefix?name:'']"
@click="handleClick">{{ icons[name] || '' }}</text>
"
`;

exports[`vue > transformCode 9`] = `
"<view class="bg-_lfl_url_lbl_https_cl__sl__sl_img_dl_cdn_dl_sugarat_dl_top_sl_mdImg_sl_MTY2ODA4OTc3MjcyMg_eqe__eqe_unocss-icon-gray_dl_svg_lbr__lfr_">
bg ==
</view>"
`;
